cores/esp32/esp32-hal-uart.c

```diff
@@ -69,6 +69,19 @@ static uart_t _uart_bus_array[] = {
 
 #endif
 
+// solves issue https://github.com/espressif/arduino-esp32/issues/6032
+// baudrate must be multiplied when CPU Frequency is lower than APB 80MHz
+uint32_t _get_effective_baudrate(uint32_t baudrate) 
+{
+    uint32_t Freq = getCpuFrequencyMhz();
```

Comment:
Maybe more clearer if you use `getApbFrequency()` which is related to UART.
But it will return frequency in Hz, not in MHz.

Reply:
Agreed here :)

Reply:
Yes, it gets better and clear. DONE. Thanks.